src/Workspaces/SharedUtilitiesAndExtensions/Compiler/CSharp/Extensions/SyntaxTreeExtensions.cs

```diff
@@ -238,8 +237,10 @@ public static bool IsEntirelyWithinCrefSyntax(this SyntaxTree syntaxTree, int po
 
             if (trivia.IsSingleLineDocComment())
             {
+                RoslynDebug.Assert(trivia.HasStructure);
+
                 var fullSpan = trivia.FullSpan;
-                var endsWithNewLine = trivia.GetStructure().GetLastToken(includeSkipped: true).Kind() == SyntaxKind.XmlTextLiteralNewLineToken;
+                var endsWithNewLine = trivia.GetStructure()!.GetLastToken(includeSkipped: true).Kind() == SyntaxKind.XmlTextLiteralNewLineToken;
```

Comment:
is there no way to tag that GetStructure() returns non-null if .HasStructure is true?

Reply:
No. `MemberNotNullWhen` only applies for properties, so this has also come up for `IsNode`/`AsNode()`.